src/Authentication/Authentication/test/Find-MgGraphPermission.Tests.ps1

```diff
@@ -2,15 +2,170 @@
+        BeforeAll {
+            Mock Invoke-MgGraphRequest {
+
+                Throw [System.Management.Automation.ValidationMetadataException]::new('mock connection error message')
+
+            }
+        }
+
+        It 'Executes successfully with no parameters' {
+            { Find-MgGraphPermission | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+
+        It "Retrieves the expected set of delegated and app-only permissions when a search string is specified" {
+            { Find-MgGraphPermission mail | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+
+        It "Returns nothing and throws no exception if a search string is specified and there is no match" {
+            { Find-MgGraphPermission 'Microsoft is the greatest company' | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+    }
+
+    Context "When executing the command after initially not having a connection to MS Graph but getting it later" {
+
+        BeforeEach {
+            Mock Invoke-MgGraphRequest {
+
+                Throw [System.Management.Automation.ValidationMetadataException]::new('mock connection error message')
+
+            }
+        }
+
         It 'Executes successfully with no parameters' {
             { Find-MgGraphPermission | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+
+            Mock Invoke-MgGraphRequest{
+                $permissionData
+            }
+
+            { Find-MgGraphPermission | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+
+        }
+
+        It "Retrieves the expected set of delegated and app-only permissions when a search string is specified" {
+            { Find-MgGraphPermission mail | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+
+            Mock Invoke-MgGraphRequest{
+                $permissionData
+            }
+
+            { Find-MgGraphPermission mail | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
         }
 
+        It "Returns nothing and throws no exception if a search string is specified and there is no match" {
+            { Find-MgGraphPermission 'Microsoft is the greatest company' | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+
+            Mock Invoke-MgGraphRequest{
+                $permissionData
+            }
+
+            { Find-MgGraphPermission 'Microsoft is the greatest company' | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
     }
 
-}
\ No newline at end of file
+    Context "When executing the command when initially not having access to read the service prinicpal but later having access" {
+
+        BeforeEach {
+            Mock Invoke-MgGraphRequest {
+
```

Comment:
For consistency you can remove the extra spaces here -- the indentions I think provide enough separation for readability.